group_vars/load-balancer/public.yml

```diff
@@ -6,8 +14,38 @@ SANITY_CHECK_LIVE_PORTS:
     port: 443
     message: "Load balancer does not respond on port 443."
 
-TARSNAP_BACKUP_FOLDERS: "/etc/letsencrypt /etc/haproxy"
+# TARSNAP #####################################################################
+
+TARSNAP_BACKUP_FOLDERS: "/etc/letsencrypt /etc/haproxy {{ consul_data_dir }}"
+
+# Consul produces consistent snapshots with these.
+TARSNAP_BACKUP_PRE_SCRIPT: "/usr/local/sbin/backup-pre-consul.sh"
+TARSNAP_BACKUP_POST_SCRIPT: "/usr/local/sbin/backup-post-consul.sh"
+
+# MAIL ########################################################################
 
 LOAD_BALANCER_OPS_EMAIL: "{{ OPS_EMAIL }}"
 
-LOAD_BALANCER_SERVER_IP: "{{ COMMON_FLOATING_IP or ansible_host }}"
+# CONSUL ######################################################################
+
+consul_backup: true
+consul_service_config:
+  services:
+    - id: haproxy-http
+      name: "haproxy"
+      tags: ["haproxy"]
+      port: 80
```

Comment:
I think haproxy should advertise the floating IP instead of the main IP of the server.  Can we inlcude something like `addr: "{{COMMON_SERVER_IP }}"` here?  (I did not check that this is the right way of doing it, nor whether `addr` and `COMMON_SERVER_IP` are correct.)

Reply:
Of course this comment only applies to the dedicated server, not to the load balancers running on VMs, so maybe we don't need this here, but rather in the private repo.

Reply:
I believe the right key is `address` and the value `{{ COMMON_FLOATING_IP }}`.

Reference: https://www.consul.io/docs/agent/services.html